### PR TITLE
[Truffle] Adding #to_int to Array#at paramters.

### PR DESCRIPTION
**Pull request by bjfish**
I removed lowerFixnumParameter because it wasn’t handling double.

I’m not sure if the guard is needed?

### CODE REVIEW

File: truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java [hunk 1]
```diff
@@ -446,10 +449,11 @@ public Object setIntegerFixnumRange(VirtualFrame frame, RubyArray array, RubyRan
 
     }
 
-    @CoreMethod(names = "at", required = 1, lowerFixnumParameters = 0)
```
Comment (bjfish) [inline truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java]: Safe to remove this?

File: truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java [hunk 2]
```diff
@@ -470,6 +474,20 @@ public Object at(VirtualFrame frame, RubyArray array, int index) {
             return readNode.executeRead(frame, array, index);
         }
 
+        @Specialization(guards = {"!isInteger(arguments[1])"})
```
Comment (bjfish) [inline truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java]: Is Guard needed?

**Comment by nirvdrum** [comment]
We could have `FixnumLowerNode` truncate, but that's probably not safe in every case.  I think you're going to need a new a `@Specialization` on `double` that does the correct conversion.  I think for the most part, Ruby truncates and then uses the resulting `Fixnum` for the index.

File: truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java (hunk 1 above)
Comment (nirvdrum) [inline truffle/src/main/java/org/jruby/truffle/nodes/core/ArrayNodes.java]: Per my comment on the PR, I think this should stay and a new specialization for `double` added.

**Pull request merged**